### ISSUE TITLE
Selecting with Shift or with Ctr key

### ISSUE THREAD

**Issue by Ofer-Gal** [monterail/vue-multiselect#495]
Not sure if there is an option to select (like in some MS controls) multiple items by clicking one then Shift click on say 3 bellow to have 4 selected at once.
Or to use the ctr key to toggle selection or not on additional items.
Please advise.
Other wise, Great project. I probably use it in production

**Comment by foRsxs** [comment]
Yeah, i wanna get same option. When you have long list and you need to selected a lot of posotions this is very VERY needed!

**Comment by shentao** [comment]
You can add the `:closeOnSelect="false"` prop that will make it easier to select multiple values in one attempt as it will prevent the multiselect from closing when selecting things. No need to additional keys to be pressed.

I think it would be quite hard to implement a selection based on the area created by moving the mouse.
I could however implement something like `shift+up` / `shift+down` to select the previous/next element. Repeat to select more elements. 
What do you think?

Also, I think the behaviour with the mouse selection area could be possible after the 3.0 rewrite.

**Comment by bibermann** [comment]
I think the most intuitive and expected gesture is Ctrl+Click which should have the effect of both `:closeOnSelect="false"` and `:clearOnSelect="false"`. This also should be the easiest to implement.

**Comment by mcmule** [comment]
bibermann, if I'm not mistaking, what you describe is already doable, using `:multiple="true"` (and `:closeOnSelect="false"`). no need to use `clearOnSelect`.

However, this is not the requested behaviour: in a lot of similar widgets (for Vue and in any software/OS really), using Shift modifier allows you to select a list of items (the list between your last selected item and the new one).

shentao: from my point of view, there is no need to use mouse area. You "just" need to keep track of last item added to selection, and when "Shift clicking" a new one, you add all items in between to the selection.
